packages/client/src/components/GrantsTable.vue

```diff
@@ -440,7 +440,7 @@ export default {
       if (!newGrantsDetailPageEnabled()) {
         return;
       }
-      this.$router.push({ name: 'grantDetail', params: { id: item.grant_id } });
+      this.$router.push({ name: 'grantDetail', params: { id: item.grant_id }, query: { fromApp: true } });
```

Comment:
We shouldn't need to surface this "from app" state in the URL where it's extra user-visible clutter. Instead, I would set up a route guard in the GrantDetails component to check the `from` route. So it can check if there's a `from` parameter, and if so, set a flag in the component data appropriately. See the docs here: https://router.vuejs.org/guide/advanced/navigation-guards.html#In-Component-Guards

Reply:
For some reason the route guard is not working. I tried the from argument and try to passed this to the `next({ source: <from code> })` but it did not worked.

Reply:
I had expected there to be a cleaner way to achieve this (i.e., `this.$router` surfacing the actual page history as a list), but alas, doesn't seem like it. I was able to get it working with something like this, though: 

```diff
diff --git a/packages/client/src/views/GrantDetails.vue b/packages/client/src/views/GrantDetails.vue
index 5b4f3100..d62c8bb5 100644
--- a/packages/client/src/views/GrantDetails.vue
+++ b/packages/client/src/views/GrantDetails.vue
@@ -173,6 +173,7 @@ export default {
   },
   data() {
     return {
+      isFirstPageLoad: false,
       showDialog: false,
       orderBy: '',
       assignedAgenciesFields: [
@@ -200,6 +201,14 @@ export default {
       copyUrlSuccessTimeout: null,
     };
   },
+  beforeRouteEnter(to, from, next) {
+    const isFirstPageLoad = from.name === null && from.path === '/';
+    next((vm) => {
+      if (isFirstPageLoad) {
+        vm.setFirstPageLoad();
+      }
+    });
+  },
   created() {
     // watch the params of the route to fetch the data again
     this.$watch(
@@ -349,6 +358,9 @@ export default {
     debounceSearchInput: debounce(function bounce(newVal) {
       this.debouncedSearchInput = newVal;
     }, 500),
+    setFirstPageLoad() {
+      this.isFirstPageLoad = true;
+    },
     async markGrantAsViewed() {
       await this.markGrantAsViewedAction({ grantId: this.selectedGrant.grant_id, agencyId: this.selectedAgencyId });
     },
```